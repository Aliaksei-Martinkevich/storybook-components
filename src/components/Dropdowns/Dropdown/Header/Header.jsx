import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Header.scss';

const Header = ({ className, children, onClick, expanded }) => (
  <div
    className={classNames('dropdown-header', className)}
    onClick={onClick}
  >
    {React.Children.map(children, child => React.cloneElement(child, { expanded }))}
  </div>
  );

Header.defaultProps = {
  className: '',
  expanded: false,
  onClick: undefined,
};

Header.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.element.isRequired,
  expanded: PropTypes.bool,
};

export default Header;
