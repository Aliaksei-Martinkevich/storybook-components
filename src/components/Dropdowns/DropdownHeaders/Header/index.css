.dropdown-header {
    margin: auto;
    min-width: 70px;
    position: relative;
    display: inline-flex;
    height: 38px;
    line-height: 38px;
}

.collapsed {
    background-image: url('../../../../../public/ico-expand-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
}

.expanded {
    background-image: url('../../../../../public/ico-collapse-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
}