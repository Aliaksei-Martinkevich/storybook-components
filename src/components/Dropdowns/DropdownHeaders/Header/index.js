import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css';

class DefaultHeader extends Component { 
  constructor(props) {
    super(props);
    this.state = {
      isExpanded: false
    };
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  
  onClickHandler(...params) {
    this.setState((prevState, props) => {
      return { isExpanded: !prevState.isExpanded };
    });
  }

  render() {
      return(
          <div className={
            `dropdown-header ${this.props.className} ${this.state.isExpanded
              ? 'expanded' 
              : 'collapsed'} `}
            onClick = {this.onClickHandler}>
            {this.props.children}
          </div>
        );
  }
}

DefaultHeader.defaultProps = {
  className: '',
}

DefaultHeader.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
}

export default DefaultHeader;
