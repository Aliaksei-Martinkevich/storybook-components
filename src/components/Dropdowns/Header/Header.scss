.dropdown-header {
    position: relative;
    display: inline-flex;
    cursor: pointer;
}
