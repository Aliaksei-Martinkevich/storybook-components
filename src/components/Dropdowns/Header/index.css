.dropdown-header {
    margin: 5px;
    position: relative;
    display: inline-flex;
    cursor: pointer;
}

/* .collapsed {
    background-image: url('../../../../../public/ico-expand-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
}

.expanded {
    background-image: url('../../../../../public/ico-collapse-arrow.png');
    background-repeat: no-repeat;
    background-position: calc(100% - 5px) center;
    background-size: 15px;
} */