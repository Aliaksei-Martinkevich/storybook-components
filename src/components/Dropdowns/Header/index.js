import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const Header = ({className, children, onClick}) => {
  return (
    <div className={`dropdown-header ${className}`}
         onClick={onClick}>
      {children}
    </div>
  );
};

Header.defaultProps = {
  className: '',
}

Header.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
}

export default Header;
