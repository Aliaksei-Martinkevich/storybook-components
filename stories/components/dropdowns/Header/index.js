import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Components from '../../../../src/components'
const Header = Components.Dropdowns.Header;
const PrimaryButton = Components.Buttons.PrimaryButton;

export default storiesOf('Dropdown Header', module)
  .add('With text', () => 
    <Header
      onClick={action('Header clicked')}>
      <span>Some text</span>
    </Header>)
  .add('With children', () => 
    <Header onClick={action('Header clicked')}>
      <div>
        Some long long long long long long text
      </div>
    </Header>)
  .add('With button', () => 
    <Header onClick={action('Header clicked')}>
      <PrimaryButton 
        onClick={action('Button clicked')} 
        text='Click me!'
      />
    </Header>);