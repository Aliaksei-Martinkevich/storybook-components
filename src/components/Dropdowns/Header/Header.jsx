import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Header.scss';

const Header = ({ className, children, onClick }) => (
  <div
    className={classNames({
      'dropdown-header': true,
      [className]: true,
    })}
    onClick={onClick}
  >
    {children}
  </div>
  );

Header.defaultProps = {
  className: '',
};

Header.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.element,
};

export default Header;
