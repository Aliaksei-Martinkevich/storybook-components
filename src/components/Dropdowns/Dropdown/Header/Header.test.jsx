import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  it('Should be collapsed and collapse it\'s children', () => {
    const customClassName = 'class-name-example';
    const component = renderer.create(
      <Header
        className={customClassName}
      >
        <div />
      </Header>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should be expanded and expand it\'s children', () => {
    const customClassName = 'class-name-example';
    const component = renderer.create(
      <Header
        className={customClassName}
        expanded
      >
        <div />
      </Header>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
