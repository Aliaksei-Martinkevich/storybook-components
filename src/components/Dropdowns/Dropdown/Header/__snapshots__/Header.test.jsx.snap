// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should be collapsed and collapse it's children 1`] = `
<div
  className="dropdown-header class-name-example"
  onClick={undefined}
>
  <div
    expanded={false}
  />
</div>
`;

exports[`Header Should be expanded and expand it's children 1`] = `
<div
  className="dropdown-header class-name-example"
  onClick={undefined}
>
  <div
    expanded={true}
  />
</div>
`;
